### PR TITLE
Add CourseUser#level_number

### PR DESCRIPTION
**Pull request by kxmbrian**

### CODE REVIEW

File: app/models/course_user.rb [hunk 1]
```diff
@@ -35,6 +35,14 @@ def experience_points
 
   delegate :experience_points, to: :experience_points_records
 
+  # Computes the level number of the CourseUser with respect to
+  # a course's Course::Levels.
+  #
+  # @return [Integer] CourseUser level number
```
Comment (lowjoel) [inline app/models/course_user.rb]: Fixnum is the type, actually.

File: spec/models/course_user_spec.rb [hunk 2]
```diff
@@ -119,6 +119,27 @@
...
+
+      context 'when student has no experience points' do
+        it { is_expected.to eq 0 }
+      end
+
+      context 'after experience points have been awarded to student' do
+        before do
+          create :course_experience_points_record, points_awarded: 150, course_user: student
+        end
+        it 'levels up' do
```
Comment (lowjoel) [inline spec/models/course_user_spec.rb]: Don't mix is_expected with an explicit message. Do:
```ruby
it { is_expected.to eq(1) }
```
Also, function calls should be parenthesised, unless it is a DSL.

File: app/models/course_user.rb [hunk 3]
```diff
@@ -35,6 +35,14 @@ def experience_points
 
   delegate :experience_points, to: :experience_points_records
 
+  # Computes the level number of the CourseUser with respect to
+  # a course's Course::Levels.
+  #
+  # @return [Fixnum] CourseUser level number
+  def level_number
```
Comment (lowjoel) [inline app/models/course_user.rb]: I didn't notice, but you should put instance methods after class instance methods (the one that goes `def self.something`)

Comment (kxmbrian) [inline app/models/course_user.rb]: Ok, re-ordered. Thanks for the feedback. I'd better re-read the style guide before each PR. haha

**Comment by lowjoel** [comment]
Final thing. Can you check if the concern will still work if you declare the methods as private?

Regardless, please squash your commits when you're done.

**Comment by kxmbrian** [comment]
The tests break when I mark them as private. Why do we want to declare them as private?

The yard type change is to correct the previous PR that this PR depends on.

**Comment by lowjoel** [comment]
Wait, yes, you can't mark them as private.

I'll take this then.

**Comment by lowjoel** [comment]
Wait a second, I just did a test. You can include the private methods of a module.

**Pull request merged**